### ISSUE TITLE
[Console] `InputInterface` extends `Stringable`

### ISSUE THREAD

**Issue by boesing** [symfony/symfony#42042]
**Description**
Hey there,

I want to create a file containing informations via `symfony/console`.
I'd like to pass the executed command to that file so it can be easily re-created later on (e.g. due to changes which have to be applied).

I've seen that `ArgvInput` as well as `ArrayInput` both already implement `__toString` method and thus, extending the `InputInterface` with the `Stringable` interface might be a solution.

**Example**
```php
interface InputOption extends Stringable
{}
```

```php
final class MyCommand extends Command
{
    public function execute(InputInterface $input, OutputInterface $output): int
    {
        $output->writeln(sprintf('You executed the following command: %s', (string) $input));
        return self::SUCCESS;
    }
}
```

Any feedback is appreciated.

**Comment by stof** [comment]
Well, extending `Stringable` in `InputInterface` is a BC break, as `__toString` is not part of the interface right now. Not sure it is worth it.

**Comment by stof** [comment]
On the other hand, as projects are not really implementing InputInterface themselves, the impact of the BC break might not be that bad and so could be done...

**Comment by boesing** [comment]
> On the other hand, as projects are not really implementing InputInterface themselves, the impact of the BC break might not be that bad and so could be done...

Yah, even tho they dont really need that feature in their own implementation, I'd be fine with having an empty string there. But when `symfony/console` is used the usual way, anything is fine.

I am aware that this is a BC break but the migration path would be quite easy for those who are affected by that change.

**Comment by wouterj** [comment]
If we want to do this, we should add `@method string __toString()` to the `InputInterface` in 5.4 first: this will trigger a deprecation if an implementation doesn't implement the method. We can then safely add the interface in 6.0.

**Comment by boesing** [comment]
I've implemented your suggestion in #42056 as an initial PoC.
